components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java

```diff
@@ -2762,71 +2762,185 @@ private boolean isFilteringNotSupported(String filterOperation) {
     }
 
     /**
-     * get the search value after appending the delimiters
-     * @param filterOperation operator value
-     * @param attributeValue search value
-     * @param delimiter delimiter for based on search type
-     * @return search attribute
+     * Get the search value after appending the delimiters according to the attribute name to be filtered.
+     *
+     * @param attributeName   Filter attribute name
+     * @param filterOperation Operator value
+     * @param attributeValue  Search value
+     * @param delimiter       Filter delimiter based on search type
+     * @return Search attribute
      */
-    private String getSearchAttribute(String filterOperation, String attributeValue, String delimiter) {
+    private String getSearchAttribute(String attributeName, String filterOperation, String attributeValue,
+            String delimiter) {
 
         String searchAttribute = null;
         if (filterOperation.equalsIgnoreCase(SCIMCommonConstants.CO)) {
-            String[] attributeItems = attributeValue.split(CarbonConstants.DOMAIN_SEPARATOR);
-            if(attributeItems.length == 2) {
-                searchAttribute = attributeItems[0] + CarbonConstants.DOMAIN_SEPARATOR +
-                        delimiter + attributeItems[1] + delimiter;
-            } else {
-                searchAttribute = delimiter + attributeValue + delimiter;
-            }
+            searchAttribute = createSearchValueForCoOperation(attributeName, filterOperation, attributeValue,
+                    delimiter);
         } else if (filterOperation.equalsIgnoreCase(SCIMCommonConstants.SW)) {
             searchAttribute = attributeValue + delimiter;
         } else if (filterOperation.equalsIgnoreCase(SCIMCommonConstants.EW)) {
+            searchAttribute = createSearchValueForEwOperation(attributeName, filterOperation, attributeValue,
+                    delimiter);
+        } else if (filterOperation.equalsIgnoreCase(SCIMCommonConstants.EQ)) {
+            searchAttribute = attributeValue;
+        }
+        return searchAttribute;
+    }
+
+    /**
+     * Create search value for CO operation.
+     *
+     * @param attributeName   Filter attribute name
+     * @param filterOperation Operator value
+     * @param attributeValue  Filter attribute value
+     * @param delimiter       Filter delimiter based on search type
+     * @return Search attribute value
+     */
+    private String createSearchValueForCoOperation(String attributeName, String filterOperation, String attributeValue,
+            String delimiter) {
+
+        // For attributes which support domain embedding, create search value by appending the delimiter after
+        // the domain separator.
+        if (isDomainSupportedAttribute(attributeName)) {
+
```

Comment:
remove extra new line

Reply:
The line after the new line is a comment. We have maintained a convention to add a new line before every comment.